arduino/builder/preprocess_sketch.go

```diff
@@ -0,0 +1,36 @@
+// This file is part of arduino-cli.
+//
+// Copyright 2023 ARDUINO SA (http://www.arduino.cc/)
+//
+// This software is released under the GNU General Public License version 3,
+// which covers the main part of arduino-cli.
+// The terms of this license can be found at:
+// https://www.gnu.org/licenses/gpl-3.0.en.html
+//
+// You can be released from the requirements of the above licenses by purchasing
+// a commercial license. Buying such a license is mandatory if you want to
+// modify or otherwise use the software for commercial activities involving the
+// Arduino software without disclosing the source code of your own applications.
+// To purchase a commercial license, send an email to license@arduino.cc.
+
+package builder
+
+import (
+	"github.com/arduino/arduino-cli/arduino/builder/preprocessor"
+	"github.com/arduino/go-paths-helper"
+)
+
+// PreprocessSketch fixdoc
+func (b *Builder) PreprocessSketch(includes paths.PathList, lineOffset int, onlyUpdateCompilationDatabase bool) error {
+	// In the future we might change the preprocessor
+	normalOutput, verboseOutput, err := preprocessor.PreprocessSketchWithCtags(
+		b.sketch, b.buildPath, includes, lineOffset,
+		b.buildProperties, onlyUpdateCompilationDatabase,
+	)
+	if b.logger.Verbose() {
+		b.logger.WriteStdout(verboseOutput)
+	}
+	b.logger.WriteStdout(normalOutput)
```

Comment:
`normalOutput` should be printed instead of `verboseOutput` not together.

Reply:
Fixed in the latest PR: 99ccfb5d4f45bda1c27c4c4b08a262a7635f7297